### PR TITLE
Set up nuget package

### PR DESCRIPTION
**Pull request by humulla**
The following task has been completed:

- [x] Create a repository for NuGet packages named lbh-core
- [x] Configure tokens to access to for GitHub packages (for publishing and consuming packages)
- [x] Create CircleCI pipelines to automate the deployment of NuGet packages.

### CODE REVIEW

File: Hackney.Core/NuGet.config [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	</config>
+	<packageSources>
+        <clear />
+	</packageSources>
+    <packageSourceCredentials>
+        <github>
+            <add key="Username" value="humulla" />
+            <add key="ClearTextPassword" value= "%NUGET_KEY%" />
+        </github>
+    </packageSourceCredentials>
```
Comment (mikemoody-amido) [inline Hackney.Core/NuGet.config]: Having this checked in (esp. with a specific non-system users username) seems wrong. Can it be removed?

File: Hackney.Core/Hackney.Core.csproj [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
```
Comment (mikemoody-amido) [inline Hackney.Core/Hackney.Core.csproj]: This project should be a class library, not an exe.

**Review by mikemoody-amido — COMMENTED** [review]
We should not be checking in a .nupkg file (Hackney.Core.1.0.0.nupkg) here.
Also does it automatically increase the version number when the package is pushed to the Nuget repository, or is that still to do?

File: Hackney.Core/Program.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+namespace Hackney.Core
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Hello World!");
+        }
+    }
+}
```
Comment (amidomichael) [inline Hackney.Core/Program.cs]: Also, this file is not needed since it is a class library

File: Hackney.Core/Hackney.Core.csproj (hunk 2 above)
Comment (humulla) [reply Hackney.Core/Hackney.Core.csproj]: This was suggested in the github documentation
[https://docs.github.com/en/packages/guides/configuring-dotnet-cli-for-use-with-github-packages#publishing-a-package-using-a-nugetconfig-file](https://docs.github.com/en/packages/guides/configuring-dotnet-cli-for-use-with-github-packages#publishing-a-package-using-a-nugetconfig-file)

File: README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+## CircleCI Pipeline 
+For NuGet Push command to work, you will need to change the version in the .csproj file according to the type of change you are making:
+
+    A specific version number is in the form Major.Minor.Patch[-Suffix], where the components have the following meanings:
+
+    Major: Breaking changes
+    Minor: New features, but backward compatible
+    Patch: Backwards compatible bug fixes only
+    Suffix (optional): a hyphen followed by a string denoting a pre-release version
```
Comment (mikemoody-amido) [inline README.md:12]: Can this not be automated as part of the pipeline?

File: Hackney.Core/Hackney.Core.csproj (hunk 2 above)
Comment (mikemoody-amido) [reply Hackney.Core/Hackney.Core.csproj]: Weird...

File: README.md (hunk 4 above)
Comment (humulla) [reply README.md:12]: This was the best solution I was able to find for the versioning unless you have any ideas?

Comment (mikemoody-amido) [reply README.md:12]: Not having any CircleCI experience, no I don't have anything better at the moment.
Hopefully we can improve it later on.

Comment (humulla) [reply README.md:12]: Definitely, this is just our first iteration so we can make lots of improvements and learn for this set up in the future

File: Hackney.Core/Hackney.Core.csproj [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (mikemoody-amido) [inline Hackney.Core/Hackney.Core.csproj]: This should be netstandard2.1. And the output type should be removed. (That github page is wrong.)

Comment (humulla) [reply Hackney.Core/Hackney.Core.csproj]: I've made this change

**Review by mikemoody-amido — APPROVED** [review]

**Pull request merged**